### PR TITLE
Raise an exception when ClassDefBuilder attempts to add a duplicate glyph.

### PR DESCRIPTION
**Pull request by tomarcher101**
There is currently an assertion in ClassDefBuild.add that stops a glyph that is already present in a class from being added again. However, there is no error reported, and thus it can be quite difficult to debug.

This PR changes this assertion into a raised exception, while also adding the related tests.

### CODE REVIEW

File: Lib/fontTools/otlLib/builder.py [hunk 1]
```diff
@@ -2574,7 +2574,8 @@ def add(self, glyphs):
             return
         self.classes_.add(glyphs)
         for glyph in glyphs:
-            assert glyph not in self.glyphs_
+            if glyph in self.glyphs_:
+                raise ValueError(f"Glyph {glyph} is already present in class.")
```
Comment (anthrotype) [inline Lib/fontTools/otlLib/builder.py:2578]: there's an `OpenTypeLibError` custom exception in `fontTools.otlLib.error` module, you should use that instead

**Comment by anthrotype** [comment]
hm, this reminded me of https://github.com/fonttools/fonttools/issues/2120

@dscorbett can you take a look?

File: Lib/fontTools/otlLib/builder.py (hunk 1 above)
Comment (tomarcher101) [reply Lib/fontTools/otlLib/builder.py]: Done. However, I'm unsure if there's anything useful I can pass in for the location?

Comment (anthrotype) [reply Lib/fontTools/otlLib/builder.py]: set location to None, it will not be printed

**Comment by anthrotype** [comment]
It's unclear to me when exactly this can occur. Could you provide a feature file snippet that reproduces this error?

**Comment by dscorbett** [comment]
I don’t remember the exact circumstances where I saw this, but I think it was in a contextual substitution rule that used a class containing the same glyph multiple times.

**Pull request merged**